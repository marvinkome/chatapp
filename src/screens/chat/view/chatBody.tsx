import * as React from 'react';
import { Avatar } from 'react-native-elements';
import { View, Text, ScrollView } from 'react-native';

import { formatDate } from '../../../lib/helpers';
import {
    chatBodyStyles as styles,
    chatBodyExtra as extraStyles
} from './styles';

const img = require('../../../../static/pp.jpg');

type Props = {
    items: Array<{
        name: string;
        text: string;
        time: any;
    }>;
};

export class ChatBody extends React.Component<Props> {
    scrollView: any;
    renderChatItem = (item: any, index: number, nextItem: any) => {
        const chatContainerStyle = [styles.chatContainer];
        const chatItemStyle = [styles.item];

        if (item.name === 'you') {
            chatContainerStyle.push(extraStyles.chatContainerMe);
            chatItemStyle.push(extraStyles.itemMe);
        } else {
            chatContainerStyle.push(extraStyles.chatContainerElse);
            chatItemStyle.push(extraStyles.itemElse);
        }

        // check if next message is from the same person
        if (nextItem && nextItem.name === item.name) {
            // join the chats
            if (nextItem.name === 'you') {
                chatItemStyle.push(extraStyles.withNextItemMe);
            } else {
                chatItemStyle.push(extraStyles.withNextItemElse);
            }
            chatContainerStyle.push(extraStyles.withNextCont);
        }

        return (
            <View style={chatContainerStyle} key={index}>
                {item.name !== 'you' && (
                    <Avatar data-testId="avatar" rounded small source={img} />
                )}

                <View style={chatItemStyle}>
                    <Text style={styles.name}>{item.name}</Text>
                    <Text style={styles.text}>{item.text}</Text>
                    <Text style={styles.time}>{formatDate(item.time)}</Text>
                </View>
            </View>
        );
    };
    newMessageIndicator = () => {
        return (
            <View style={styles.newMsg}>
                <Text style={styles.newMsgText}>New Message</Text>
            </View>
        );
    };
    render() {
        return (
            <ScrollView
                style={styles.container}
                ref={(ref: any) => (this.scrollView = ref)}
                onContentSizeChange={(contentWidth, contentHeight) => {
                    this.scrollView.scrollToEnd({ animated: false });
                }}
            >
                {this.props.items.map((item, index, array) =>
                    this.renderChatItem(item, index, array[index + 1])
                )}
            </ScrollView>
        );
    }
}
