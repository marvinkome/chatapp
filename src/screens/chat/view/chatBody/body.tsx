import * as React from 'react';
import _ from 'lodash';
import moment from 'moment';

// UI Elements
import { FlatList } from 'react-native';
import { List } from 'react-native-elements';
import Hr from '../../../../components/Hr';
import { ChatMsg } from './listItem';

// styles
import { chatMsg as style } from './styles';

export interface Props {
    items: Array<{
        id: string;
        from: {
            id: string;
            username: string;
        };
        message: string;
        timestamp: any;
    }>;
    subscribe: () => void;
}

function RenderItem(props: any) {
    return props.item.messages.map((item: any) => (
        <ChatMsg key={item.id} {...item} />
    ));
}

export function ChatBody(props: Props) {
    const data = _.chain(props.items)
        .groupBy((item) => new Date(Number(item.timestamp)).toDateString())
        .map((messages, timestamp) => ({ messages, timestamp }))
        .value();

    return (
        <List containerStyle={style.listContainer}>
            <FlatList
                data={data}
                renderItem={RenderItem}
                ItemSeparatorComponent={({ leadingItem }) => (
                    <Hr
                        text={moment(
                            leadingItem.timestamp,
                            'ddd MMM DD YYYY'
                        ).format('D MMM Y')}
                    />
                )}
                keyExtractor={(item) => item.timestamp}
                inverted
            />
        </List>
    );
}
