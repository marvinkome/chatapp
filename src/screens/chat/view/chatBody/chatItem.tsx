import React from 'react';
import { Avatar } from 'react-native-elements';
import { View, Text } from 'react-native';

import { formatDate } from '../../../../lib/helpers';
import {
    chatBodyStyles as styles,
    chatBodyExtra as extraStyles
} from '../styles';

const img = require('../../../../../static/pp.jpg');

interface Props {
    item: {
        from: {
            id: string;
            username: string;
        };
        message: string;
        timestamp: any;
    };
    nextItem: {
        from: {
            id: string;
            username: string;
        };
        message: string;
        timestamp: any;
    };
    currentUser: string;
}

function ChatView({ item, nextItem, currentUser }: Props) {
    const chatContainerStyle = [styles.chatContainer];
    const chatItemStyle = [styles.item];

    if (item.from.username === currentUser) {
        chatContainerStyle.push(extraStyles.chatContainerMe);
        chatItemStyle.push(extraStyles.itemMe);
    } else {
        chatContainerStyle.push(extraStyles.chatContainerElse);
        chatItemStyle.push(extraStyles.itemElse);
    }

    // check if next message is from the same person
    if (nextItem && nextItem.from.username === item.from.username) {
        // join the chats
        if (nextItem.from.username === currentUser) {
            chatItemStyle.push(extraStyles.withNextItemMe);
        } else {
            chatItemStyle.push(extraStyles.withNextItemElse);
        }
        chatContainerStyle.push(extraStyles.withNextCont);
    }

    return (
        <View style={chatContainerStyle}>
            {item.from.username !== currentUser && (
                <Avatar data-testId="avatar" rounded small source={img} />
            )}

            <View style={chatItemStyle}>
                {item.from.username !== currentUser && (
                    <Text style={styles.name}>{item.from.username}</Text>
                )}
                <Text style={styles.text}>{item.message}</Text>
                <Text style={styles.time}>
                    {formatDate(Number(item.timestamp))}
                </Text>
            </View>
        </View>
    );
}

export default ChatView;
