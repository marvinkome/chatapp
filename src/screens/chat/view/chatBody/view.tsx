import * as React from 'react';
import { View, Text, ScrollView } from 'react-native';
import ChatItem from './chatItem';
import { chatBodyStyles as styles } from '../styles';

interface Props {
    loggedUser: string;
    items: Array<{
        id: string;
        from: {
            id: string;
            username: string;
        };
        message: string;
        timestamp: any;
    }>;
}

export default class ChatBody extends React.Component<Props> {
    scrollView: any;
    newMessageIndicator = () => {
        return (
            <View style={styles.newMsg}>
                <Text style={styles.newMsgText}>New Message</Text>
            </View>
        );
    };
    render() {
        return (
            <ScrollView
                style={styles.container}
                ref={(ref: any) => (this.scrollView = ref)}
                onContentSizeChange={() => {
                    this.scrollView.scrollToEnd({ animated: false });
                }}
            >
                {this.props.items.map((item, index, array) => (
                    <ChatItem
                        item={item}
                        nextItem={array[index + 1]}
                        currentUser={this.props.loggedUser}
                        key={item.id}
                    />
                ))}
            </ScrollView>
        );
    }
}
