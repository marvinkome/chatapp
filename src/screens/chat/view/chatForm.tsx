import * as React from 'react';
import { View } from 'react-native';
import { FormInput, Button } from 'react-native-elements';
import { chatFormStyles as styles } from './styles';

type IProps = {
    sendMessage: (data: string) => void;
};

type IState = {
    message: string;
};

export class ChatForm extends React.Component<IProps, IState> {
    state = {
        message: ''
    };

    onMessageChange = (message: string) => {
        this.setState({
            message
        });
    };

    sendMessage = () => {
        this.props.sendMessage(this.state.message);
    };

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.formContainer}>
                    <FormInput
                        placeholder="Write a message..."
                        underlineColorAndroid="transparent"
                        inputStyle={styles.input}
                        onChangeText={this.onMessageChange}
                        multiline
                    />
                </View>
                <View style={styles.btnCont}>
                    <Button
                        iconRight={{ name: 'md-send', type: 'ionicon' }}
                        title="BTN"
                        onPress={this.sendMessage}
                        containerViewStyle={styles.btn}
                        buttonStyle={styles.btnStyle}
                        textStyle={styles.btnText}
                        large
                    />
                </View>
            </View>
        );
    }
}
