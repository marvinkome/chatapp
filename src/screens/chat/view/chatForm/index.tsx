import * as React from 'react';
import { View, Keyboard } from 'react-native';
import { FormInput, Button } from 'react-native-elements';
import { withApollo, WithApolloClient } from 'react-apollo';
import { chatFormStyles as styles } from '../styles';

import { userTyping } from './gql';

type IProps = WithApolloClient<{
    sendMessage: (data: string) => void;
    groupId: string;
}>;

type IState = {
    message: string;
};

class ChatForm extends React.Component<IProps, IState> {
    keyboardDidShowListener: any;
    keyboardDidHideListener: any;

    state = {
        message: ''
    };

    componentDidMount() {
        this.keyboardDidShowListener = Keyboard.addListener(
            'keyboardDidShow',
            this.triggerUserTyping
        );
        this.keyboardDidHideListener = Keyboard.addListener(
            'keyboardDidHide',
            this.triggerUserNotTyping
        );
    }

    componentWillUnmount() {
        this.keyboardDidHideListener.remove();
        this.keyboardDidShowListener.remove();
    }

    triggerUserTyping = () => {
        this.props.client.mutate({
            mutation: userTyping,
            variables: {
                groupId: this.props.groupId
            }
        });
    };
    triggerUserNotTyping = () => null;

    onMessageChange = (message: string) => {
        this.setState({
            message
        });
    };

    sendMessage = () => {
        this.props.sendMessage(this.state.message);
        this.setState({
            message: ''
        });
    };

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.formContainer}>
                    <FormInput
                        placeholder="Write a message..."
                        underlineColorAndroid="transparent"
                        inputStyle={styles.input}
                        onChangeText={this.onMessageChange}
                        value={this.state.message}
                        multiline
                    />
                </View>
                <View style={styles.btnCont}>
                    <Button
                        iconRight={{ name: 'md-send', type: 'ionicon' }}
                        title="BTN"
                        onPress={this.sendMessage}
                        containerViewStyle={styles.btn}
                        buttonStyle={styles.btnStyle}
                        textStyle={styles.btnText}
                        large
                    />
                </View>
            </View>
        );
    }
}

export default withApollo(ChatForm);
