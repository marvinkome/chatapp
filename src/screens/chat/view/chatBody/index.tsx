import * as React from 'react';
import { Subscription, SubscriptionResult } from 'react-apollo';

// UI Element
import { ChatBody } from './body';

// graphql documents
import { userTypingEvent } from './gql';

export interface Props {
    items: Array<{
        id: string;
        from: {
            id: string;
            username: string;
        };
        message: string;
        timestamp: any;
    }>;
    groupId: string;
    subscribe: () => void;
}

export default class MainChatBody extends React.Component<Props> {
    componentDidMount() {
        // subscribe to more messages
        this.props.subscribe();
    }

    render() {
        const subscriptionProps = {
            subscription: userTypingEvent,
            variables: { groupId: this.props.groupId },
            fetchPolicy: 'network-only'
        };

        return (
            <Subscription {...subscriptionProps}>
                {({ data }) => {
                    return <ChatBody typingData={data} {...this.props} />;
                }}
            </Subscription>
        );
    }
}
