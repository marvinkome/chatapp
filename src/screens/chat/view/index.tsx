import * as React from 'react';
import { View } from 'react-native';

import Socket from '../../../lib/socket';
import { ChatHeader } from './chatHeader';
import { ChatBody } from './chatBody';
import { ChatForm } from './chatForm';
import { viewStyles as styles } from './styles';

type IState = {
    messages: Array<{
        name: string;
        text: string;
        time: any;
    }>;
};

class ChatView extends React.Component<{}, IState> {
    socket: SocketIOClient.Socket;
    state = {
        messages: []
    };

    componentDidMount() {
        this.socket = Socket();

        this.socket.on('message response', (data: any) => {
            this.setState({
                messages: this.state.messages.concat(data)
            });
        });
    }

    sendMessage = (message: string) => {
        this.socket.emit('send message', message);
    };

    render() {
        return (
            <View style={styles.container}>
                <ChatHeader data-testId="chat-header" />
                <ChatBody data-testId="chat-body" items={this.state.messages} />
                <ChatForm sendMessage={this.sendMessage} />
            </View>
        );
    }
}

export default ChatView;
