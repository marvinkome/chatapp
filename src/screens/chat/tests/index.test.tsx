import React from 'react';
import Chats from '../index';
import View from '../view';
import { ChatBody } from '../view/chatBody';
import { ChatForm } from '../view/chatForm';
import { create } from 'react-test-renderer';

describe('chat screen', () => {
    it('renders correctly', () => {
        const comp = create(<Chats />);
        expect(comp.toJSON()).toMatchSnapshot();
    });

    describe('chat view', () => {
        const view = create(<View />);
        const viewIns = view.root;
        it('renders (snapshot)', () => {
            expect(view.toJSON()).toMatchSnapshot();
        });

        // should containing the to - user name
        it('should contain the reciever username and last seen', () => {
            const comp = viewIns.findByProps({
                ['data-testId']: 'chat-header'
            });
            const name = comp.findByProps({ ['data-testID']: 'username' });
            const lastSeen = comp.findByProps({ ['data-testID']: 'last-seen' });

            expect(name.props.children).toBe('Larry Doe');
            expect(lastSeen.props.children).toBe('Last seen: 12/12/17');
        });

        describe('chat body', () => {
            const listItems = [
                {
                    name: 'jamesbond',
                    text: 'Lorem ipsum dolor sit amet consectetur',
                    time: '9 Aug 2018 23:00'
                },
                {
                    name: 'you',
                    text: 'Doing?',
                    time: 'Today at 13:45'
                }
            ];
            it('renders 4 chats', () => {
                const chatItems = create(<ChatBody items={listItems} />);
                expect(chatItems).toMatchSnapshot();
            });

            // should have avatar component only on messages from sender
            // only first message should have avatar image when there's multple messages
            // there should be a new messages indicator
        });

        describe('chat form', () => {
            it('should render correctly', () => {
                const comp = create(<ChatForm />);
                expect(comp).toMatchSnapshot();
            });
        });
    });
});
