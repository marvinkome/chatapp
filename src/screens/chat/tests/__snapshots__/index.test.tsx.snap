// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat screen chat view chat body renders 4 chats 1`] = `
<RCTScrollView
  onContentSizeChange={[Function]}
  style={
    Object {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 15,
            "width": "80%",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        data-testId="avatar"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 34,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "width": 34,
            },
            Object {
              "borderRadius": 17,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "rgba(0,0,0,0.2)",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 17,
              },
              undefined,
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../static/pp.jpg",
              }
            }
            style={
              Array [
                Object {
                  "height": 34,
                  "width": 34,
                },
                Object {
                  "borderRadius": 17,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#42ED86",
              "borderRadius": 10,
              "padding": 10,
            },
            Object {
              "borderTopLeftRadius": 0,
              "marginLeft": 13,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 12,
              "marginBottom": 7,
              "opacity": 0.8,
            }
          }
        >
          jamesbond
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "PT_Sans",
              "fontSize": 16,
              "letterSpacing": -0.25,
            }
          }
        >
          Lorem ipsum dolor sit amet consectetur
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 12,
              "marginTop": 7,
              "opacity": 0.8,
            }
          }
        >
          a few seconds ago
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 15,
            "width": "80%",
          },
          Object {
            "alignSelf": "flex-end",
            "flexDirection": "row-reverse",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#42ED86",
              "borderRadius": 10,
              "padding": 10,
            },
            Object {
              "borderTopRightRadius": 0,
              "marginRight": 13,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 12,
              "marginBottom": 7,
              "opacity": 0.8,
            }
          }
        >
          you
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "PT_Sans",
              "fontSize": 16,
              "letterSpacing": -0.25,
            }
          }
        >
          Doing?
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 12,
              "marginTop": 7,
              "opacity": 0.8,
            }
          }
        >
          a few seconds ago
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`chat screen chat view chat form should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 4,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Write a message..."
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "height": 65,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#42ED86",
                "paddingLeft": 7,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  false,
                  undefined,
                  Object {
                    "display": "none",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            BTN
          </Text>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 26,
                },
                Array [
                  Object {
                    "marginLeft": 10,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            type="ionicon"
          >
            Ôç∂
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`chat screen chat view renders (snapshot) 1`] = `
<View
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 96%)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#476DC5",
          "borderBottomColor": "#f2f2f2",
          "borderBottomWidth": 1,
          "height": 70,
          "padding": 15,
        },
        undefined,
        Object {
          "backgroundColor": "#fff",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "#42ED86",
                    "fontSize": 37,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ÔÑó
          </Text>
        </View>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          data-testID="username"
          ellipsizeMode="tail"
          style={
            Object {
              "color": "hsl(0, 0%, 20%)",
              "fontFamily": "Karla",
              "fontSize": 19,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Larry Doe
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          data-testID="last-seen"
          ellipsizeMode="tail"
          style={
            Object {
              "color": "hsl(0, 0%, 45%)",
              "fontFamily": "Karla",
              "fontSize": 14,
              "textAlign": "center",
            }
          }
        >
          Last seen: 12/12/17
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    onContentSizeChange={[Function]}
    style={
      Object {
        "flex": 1,
        "padding": 16,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 4,
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomColor": "#bdc6cf",
            "borderBottomWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Write a message..."
          style={
            Array [
              Object {
                "color": "#86939e",
                "minHeight": 36,
                "width": 750,
              },
              Object {
                "fontSize": 17.5,
              },
              Object {
                "height": 65,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                false,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#42ED86",
                  "paddingLeft": 7,
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    false,
                    undefined,
                    Object {
                      "display": "none",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              BTN
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 26,
                  },
                  Array [
                    Object {
                      "marginLeft": 10,
                    },
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="ionicon"
            >
              Ôç∂
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`chat screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 96%)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#476DC5",
          "borderBottomColor": "#f2f2f2",
          "borderBottomWidth": 1,
          "height": 70,
          "padding": 15,
        },
        undefined,
        Object {
          "backgroundColor": "#fff",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "#42ED86",
                    "fontSize": 37,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ÔÑó
          </Text>
        </View>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          data-testID="username"
          ellipsizeMode="tail"
          style={
            Object {
              "color": "hsl(0, 0%, 20%)",
              "fontFamily": "Karla",
              "fontSize": 19,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Larry Doe
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          data-testID="last-seen"
          ellipsizeMode="tail"
          style={
            Object {
              "color": "hsl(0, 0%, 45%)",
              "fontFamily": "Karla",
              "fontSize": 14,
              "textAlign": "center",
            }
          }
        >
          Last seen: 12/12/17
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    onContentSizeChange={[Function]}
    style={
      Object {
        "flex": 1,
        "padding": 16,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 4,
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomColor": "#bdc6cf",
            "borderBottomWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Write a message..."
          style={
            Array [
              Object {
                "color": "#86939e",
                "minHeight": 36,
                "width": 750,
              },
              Object {
                "fontSize": 17.5,
              },
              Object {
                "height": 65,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                false,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#42ED86",
                  "paddingLeft": 7,
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    false,
                    undefined,
                    Object {
                      "display": "none",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              BTN
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 26,
                  },
                  Array [
                    Object {
                      "marginLeft": 10,
                    },
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="ionicon"
            >
              Ôç∂
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
